lib/file.c

```diff
@@ -400,6 +406,64 @@ static CURLcode file_upload(struct Curl_easy *data)
   return result;
 }
 
+/*
+ * directory_list() will read all entries in a directory and prepare
+ * a temporary char * buffer with the full list, with entries separated
+ * by \n, for file_do().
+ */
+static CURLcode directory_list(struct FILEPROTO *file, char **data, curl_off_t *size) {
+  CURLcode result = CURLE_OK;
+  DIR *dir;
+  struct dirent *entry;
+  curl_off_t entry_len;
+
+  *data = NULL;
+  *size = 0;
+
+  if(!file) {
+    result = CURLE_BAD_FUNCTION_ARGUMENT;
+    goto out;
+  }
+
+  dir = fdopendir(file->fd);
+  if(!dir) {
+    result = CURLE_BAD_FUNCTION_ARGUMENT;
```

Comment:
This error is curious, since it already opened the file and it is said to be a dir, but this still fails. It seems to me that `CURLE_BAD_FUNCTION_ARGUMENT` is not the most accurate description of this error. Maybe just `CURLE_READ_ERROR` ?